### ISSUE TITLE
ECMA402 abstract methods

### ISSUE THREAD

**Issue by longlho** [ljharb/es-abstract#109]
Hi there,

We wrote a lot of ECMA402 abstract ops for formatjs and I'm curious if the maintainers would be open to us moving things over here?

Thanks!

**Comment by ljharb** [comment]
That's a great question.

Currently it's only 262 methods, and it's split out by year. If we added 402, I'd want to also go back and add them for previous years.

In v2, the "manifest" exports will all be removed, so there wouldn't possibly be any bundle size increase from adding these - but in v1, every addition does add a cost for anyone importing an `ES` object.

If you'd be willing, sending a PR for all of the current ones would be great! However, instead of mixing them with 262 operations (since 402 also redefines some of them), what about adding an `ecma402` top-level directory, and adding year folders in there?

**Comment by longlho** [comment]
kk I'll give that a try

**Comment by mpcsh** [comment]
@longlho would you be interested in coordinating on some of this work? I wrote [half a polyfill](https://github.com/mpcsh/proposal-intl-segmenter-review) of `Intl.Segmenter` as part of my stage 3 review; to turn it into an actual polyfill, I need to add a bunch of 402 methods to `es-abstract`. I'd love to work together!

**Comment by longlho** [comment]
Yeah definitely ;) some project management help to plan out the work & ofc actual contribution would be awesome!!

**Comment by mpcsh** [comment]
let's talk - are you on IRC? I don't see anyone with your username. alternately, feel free to DM me on [twitter](https://twitter.com/messages/compose?recipient_id=1171910612269654018).

**Comment by longlho** [comment]
Btw adding 402 from previous years would be a huge amount of maintenance from what I've seen. Since ECMA402 correctness is "best-effort" and a ton of things are ILND (implementation-locale-numberingSystem-dependent) some stuff would have to be completely rewritten. For example: the algorithm to parse & validate locale changed recently in ES2020 where it used to be BCP47, now it's canonicalized UTS35 BCP47 which is a different algorithm.

**Comment by ljharb** [comment]
Sure, but we can start with just 2020, make it easy to add 2021, and then someone who's obsessive about completeness (hi!) can easily fill in older years over time.
